woocommerce/woocommerce-android
# Display quantity rules in product details

⚠️ This PR depends on [this FluxC PR](https://github.com/wordpress-mobile/WordPress-FluxC-Android/pull/2697)

Closes: #8400
Closes: #8401

### Why
We are adding read-only support to the Min/Max Quantities extension

### Description
This PR adds the necessary changes to display the product's quantity rules on the product details screen. It does that by adding the new `quantityRules` extension function to the `ProductDetailCardBuilder` and the `ProductQuantityRulesFragment` to display the quantity rules details. The `GetProductQuantityRules` is used to get the quantity rules depending on the availability of the Min / Max extension. In this first iteration, the Quantity Rules card is not shown if the quantity rules are empty, we might want to change this behavior in the future when adding editing support. 

### Testing instructions
TC1
1. Open the Products tab
2. Select any product with quantity rules
3. Check that the Quantity Rules card is shown
4. Tap on the Quantity Rules card
5. Check that the app navigates to the Quantity Rules fragment
6. Press back
7. Check that the app returns to the product details screen

TC2
1. Open the Products tab
2. Select any product without quantity rules
3. Check that the Quantity Rules card is NOT shown

### Images/gif

https://user-images.githubusercontent.com/18119390/229360695-c9ae0904-2e07-4cf5-ab1c-6bccfc045995.mp4
